### ISSUE TITLE
config:set cli not saving config value to default scope

### ISSUE THREAD

**Issue by ghost** [magento/magento2#23669]
### Preconditions (*)

1. Magento version 2.2.9
2.

### Steps to reproduce (*)

1. on a Magento instance running in single store mode, use the cli to set the paypal business account with `magento config:set paypal/general/business_account 'paypal@example.com'`

### Expected result (*)

1. the value paypal@example.com should be saved in core_config_data table with scope set to 'default', but scope is set to 'websites'. Furthermore, the scope id is set to '1' instead of '0'
2. Furthermore, the newly saved value should be visible in Stores -> Config -> Sales -> Payment Methods -> Paypal Field: Email Associated with PayPal Merchant Account (Optional)

### Actual result (*)

1. The value is saved in core_config_data in the websites scope 
2. no value is visible in Stores -> Config -> Sales -> Payment Methods -> Paypal Field: Email Associated with PayPal Merchant Account (Optional)

From what I have seen, in order for the config:set command to work, the corresponding system.xml has to define the field in a specific way. In this example, the field's corresponding section id must be 'paypal', the group id must be 'general', and the field id must be 'business_account'. The config_path is not honored. I haven't debugged far enough to provide more info.

This stopped working when we upgraded from 2.2.7 to 2.2.9.

**Comment by ghost** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shilpambb. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shilpambb, here is your [Magento instance](https://i-23669-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23669-2-3-develop.instances.magento-community.engineering/admin](https://i-23669-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ghost** [comment]
I'm not sure how to gain cli access to reproduce the issue. Any suggestions?

**Comment by engcom-Bravo** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Bravo, here is your [Magento instance](https://i-23669-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23669-2-3-develop.instances.magento-community.engineering/admin](https://i-23669-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Bravo** [comment]
Hi @shilpambb thank you for your report, I'm not able to reproduce issue by steps you described on clean 2.3-develop. 

![v6](https://user-images.githubusercontent.com/51680745/61060970-158e9100-a404-11e9-8d4c-c93c779d1602.png)
![v7](https://user-images.githubusercontent.com/51680745/61060971-158e9100-a404-11e9-996d-cb53ad1ab372.png)

**Comment by ghost** [comment]
hmmm, ok, I'll look into my environment some more.

**Comment by ghost** [comment]
Can we try this on a 2.2.9 instance? I disabled all my third party modules and still can't get it to work on a 2.2.9 instance. Also, if there is a way, I'd like cli access on one of the vanilla instances to try it out.

**Comment by shilpambb** [comment]
Can we test this out on 2.2.9? I'm able to reproduce it locally after disabling all third party modules.

**Comment by hostep** [comment]
Reopening, @engcom-Bravo, you can very easily reproduce this problem on the `2.3-develop` branch.
Make sure you enabled single store mode like @shilpambb mentioned in the opening post.

<img width="1318" alt="Screenshot 2019-07-12 at 20 20 22" src="https://user-images.githubusercontent.com/85479/61149966-a20f8100-a4e2-11e9-84b3-8f4424e056e5.png">

@shilpambb: thanks for creating this issue, it does sound like a bug indeed!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18160` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Bravo** [comment]
@shilpambb @hostep , thanks for communication!

**Comment by lurajcevi** [comment]
This is reproducible on `2.4-develop` as well.

After looking into the code, when `Single store mode` is enabled, there is a special check in the code at: `\Magento\Config\Model\Config::_processGroup`, [L454](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Config/Model/Config.php#L454) calling the method: `\Magento\Config\Model\Config::_checkSingleStoreMode` which is responsible for changing the scope of config entry if single store mode is enabled **and** `showInDefault`  of the config is false.

This is why scope is changed from `default` to `websites`.

```
    protected function _checkSingleStoreMode(
        \Magento\Config\Model\Config\Structure\Element\Field $fieldConfig,
        $dataObject
    ) {
        $isSingleStoreMode = $this->_storeManager->isSingleStoreMode();
        if (!$isSingleStoreMode) {
            return;
        }
        if (!$fieldConfig->showInDefault()) {
            $websites = $this->_storeManager->getWebsites();
            $singleStoreWebsite = array_shift($websites);
            $dataObject->setScope('websites');
            $dataObject->setWebsiteCode($singleStoreWebsite->getCode());
            $dataObject->setScopeCode($singleStoreWebsite->getCode());
            $dataObject->setScopeId($singleStoreWebsite->getId());
        }
    }
```

So, I'm assuming this is the correct behaviour.

As for why this happens for this specific path is because `paypal/general/business_account` does not have an entry in any of system xml files (`system.xml`, or any other in `Paypal/etc/adminhtml/system/` directory) - at least I wasn't able to find one, and `showInDefault` is not present when config is loaded, thus is `false`, and the condition logic is evaluated.

**Comment by magento-engcom-team** [comment]
Hi @ghost. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-23833` was closed as `non-reproducible` in `2.4-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.

**Comment by engcom-Alfa** [comment]
The issue isn't reproducible on a fresh 2.4-develop.

**Actual Result:** 

:heavy_check_mark:   The value is saved in core_config_data with scope set to 'default' ( scope_id = 0)

![2021-02-22_14-06](https://user-images.githubusercontent.com/51679138/108708210-d4d08680-7519-11eb-8b88-71e1a6abd5fb.png)


:heavy_check_mark:   the newly saved value is visible in Stores -> Config -> Sales -> Payment Methods -> Paypal Field: Email Associated with PayPal Merchant Account (Optional)

![2021-02-22_14-23](https://user-images.githubusercontent.com/51679138/108708218-d7cb7700-7519-11eb-8428-4aba0a49f820.png)

**Comment by magento-engcom-team** [comment]
Hi @ghost. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-23833` was closed as `non-reproducible` in `2.4-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.
